skyplane/obj_store/s3_interface.py

```diff
@@ -63,8 +72,15 @@ def delete_bucket(self):
     def list_objects(self, prefix="") -> Iterator[S3Object]:
         paginator = self._s3_client().get_paginator("list_objects_v2")
         page_iterator = paginator.paginate(Bucket=self.bucket_name, Prefix=prefix)
-        for page in page_iterator:
-            for obj in page.get("Contents", []):
+        if not self.requester_pays:
```

Comment:
You should be able to set the RequesterPays using this method: https://stackoverflow.com/a/44376294